pkg/docker/config/config.go

```diff
@@ -44,6 +46,11 @@ func SetAuthentication(sys *types.SystemContext, registry, username, password st
 			return false, setAuthToCredHelper(ch, registry, username, password)
 		}
 
+		err := setAuthToKernelKeyring(registry, username, password)
```

Comment:
Can we add a `logrus.Debug()` here to indicate what's happening? We just recently added some logging for where the creds come from, and I think it can be useful here as well.

Reply:
The call to `setAuthToKernelKeyring(...)` should be moved to `GetAuthentication(...)`. `findAuthentication(...)` is meant for lookups in specific files. At the moment, we contact the keyring for each item in the slice of paths.

Reply:
And lets add a comment explaining the fall-through to using the authfiles, in case an error occured (and `logrus.Debug(...)` the error).

Reply:
I think you mean that I should move `getAuthFromKernelKeyring`, not `setAuthToKernelKeyring(...)` :slightly_smiling_face: 

> The call to `setAuthToKernelKeyring(...)` should be moved to `GetAuthentication(...)`. `findAuthentication(...)` is meant for lookups in specific files. At the moment, we contact the keyring for each item in the slice of paths.

Reply:
:rofl: wow, I scrolled and copied the wrong one. Yes, absolutely, it is `getAuthFromKernelKeyring` what I meant.